client/web-sveltekit/src/lib/search/dynamicFilters/Sidebar.svelte

```diff
@@ -1,151 +1,210 @@
+            return 'Linux'
+        }
+
+        return undefined
+    }
+</script>
+
 <script lang="ts">
-    import { mdiBookOpenVariant } from '@mdi/js'
+    import { onDestroy, onMount } from 'svelte'
+
+    import type { Filter as QueryFilter } from '@sourcegraph/shared/src/search/query/token'
 
+    import { goto } from '$app/navigation'
+    import { page } from '$app/stores'
     import Icon from '$lib/Icon.svelte'
+    import ArrowBendIcon from '$lib/icons/ArrowBend.svelte'
+    import LanguageIcon from '$lib/LanguageIcon.svelte'
     import CodeHostIcon from '$lib/search/CodeHostIcon.svelte'
-    import Section from '$lib/search/dynamicFilters/Section.svelte'
-    import TypeSection from '$lib/search/dynamicFilters/TypeSection.svelte'
-    import type { QueryStateStore } from '$lib/search/state'
     import SymbolKind from '$lib/search/SymbolKind.svelte'
-    import { groupFilters } from '$lib/search/utils'
-    import { displayRepoName, type Filter } from '$lib/shared'
+    import { displayRepoName, scanSearchQuery, type Filter } from '$lib/shared'
+    import Tooltip from '$lib/Tooltip.svelte'
+    import Button from '$lib/wildcard/Button.svelte'
 
-    export let queryFromURL: string
+    import HelpFooter from './HelpFooter.svelte'
+    import {
+        type URLQueryFilter,
+        type SectionItem,
+        staticTypeFilters,
+        typeFilterIcons,
+        groupFilters,
+        moveFiltersToQuery,
+        resetFilters,
+    } from './index'
+    import LoadingSkeleton from './LoadingSkeleton.svelte'
+    import Section from './Section.svelte'
+
+    export let searchQuery: string
+    export let loading: boolean
     export let streamFilters: Filter[]
-    export let queryFilters: string
-    export let queryState: QueryStateStore
-    export let size: number
+    export let selectedFilters: URLQueryFilter[]
+
+    $: groupedFilters = groupFilters(streamFilters, selectedFilters)
+    $: typeFilters = staticTypeFilters.map((staticTypeFilter): SectionItem => {
+        const selectedOrStreamFilter = groupedFilters.type.find(
+            typeFilter => typeFilter.label === staticTypeFilter.label
+        )
+        return {
+            ...staticTypeFilter,
+            count: selectedOrStreamFilter?.count,
+            exhaustive: selectedOrStreamFilter?.exhaustive || false,
+            selected: selectedOrStreamFilter?.selected || false,
+        }
+    })
 
-    $: width = `max(100px, min(50%, ${size * 100}%))`
-    $: filters = groupFilters(streamFilters)
-    $: hasFilters = filters.lang.length > 0 || filters.repo.length > 0 || filters.file.length > 0
+    $: resetModifier = inferOperatingSystem(navigator.userAgent) === 'MacOS' ? '⌥' : 'Alt'
+    $: resetURL = resetFilters($page.url).toString()
+    $: enableReset = selectedFilters.length > 0
+    $: handleResetKeydown = (event: KeyboardEvent) => {
+        if (enableReset && event.altKey && event.key === 'Backspace') {
+            goto(resetURL)
+        }
+    }
```

Comment:
This doesn't have to be reactive. The function will close over `enableReset` and `resetURL` and use whatever values those variables have when the function is called.

```suggestion
    function handleResetKeydown(event: KeyboardEvent) {
        if (enableReset && event.altKey && event.key === 'Backspace') {
            goto(resetURL)
        }
    }
```